# Results

In [ ]:
from pathlib import Path
import pandas as pd
from src.helpers.metrics import  build_metrics_table

In [2]:
data_name = "DS03"
degmodel_name = 'normal'
filtered_data = True
pfnet_dir_name = 'net256x256x128x64x32leaky0.05'
n_rep = 10

filtered_text = "_filtered" if filtered_data else ""
experiment_dir = Path('experiments')/data_name
pfnet_dir = experiment_dir/f'degradation{filtered_text}'/degmodel_name/'pf_performs'/pfnet_dir_name

## Load Predictions

In [3]:
df_preds = pd.read_csv(pfnet_dir / f"rul_test_predictions_rep{n_rep}.csv")

## Mean Prediction

In [4]:
df_preds = df_preds.groupby(["unit", "time"], as_index=False).mean()

## Degradation Predictions

In [5]:
hi_df = pd.read_csv(experiment_dir/f'hidata_test{filtered_text}.csv')
test_units = hi_df['unit'].astype(int).unique().tolist()
onsets = {unit: hi_df[(hi_df['unit']==unit) & (hi_df['hs']==0)]['cycle'].values[0] for unit in test_units}
df_degpreds = df_preds[
    df_preds["time"] > df_preds["unit"].map(onsets)
]

## Metrics

In [6]:
metrics_df = build_metrics_table(df_preds)
metrics_df

unit       RMSE       Score      PICP     PINAW
0    10   4.190528   26.301490  1.000000  0.509070
1    11  10.147794   95.884897  1.000000  0.586171
2    12  13.850840  170.770287  0.913043  0.351767
3    13   5.661497   33.993773  0.960526  0.406547
4    14   2.815029   13.459270  1.000000  0.461663
5    15   4.812033   31.726726  1.000000  0.508752

In [7]:
metrics_df.drop(columns=["unit"]).mean()

RMSE      6.912953
Score    62.022741
PICP      0.978928
PINAW     0.470662
dtype: float64

In [8]:
metrics_df = build_metrics_table(df_degpreds)
metrics_df

unit      RMSE      Score      PICP     PINAW
0    10  2.287758   8.309579  1.000000  0.549712
1    11  5.867247  24.833282  1.000000  0.640541
2    12  7.713505  38.687416  0.982143  0.343562
3    13  6.360628  30.876775  0.948276  0.417969
4    14  1.265477   4.052291  1.000000  0.480910
5    15  2.201658   5.400241  1.000000  0.516345

In [9]:
metrics_df.drop(columns=["unit"]).mean()

RMSE      4.282712
Score    18.693264
PICP      0.988403
PINAW     0.491506
dtype: float64

# DS03 (normal-filtered-full life)
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>unit</th>
      <th>RMSE</th>
      <th>Score</th>
      <th>PICP</th>
      <th>PINAW</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>10</td>
      <td>4.190528</td>
      <td>26.301490</td>
      <td>1.000000</td>
      <td>0.509070</td>
    </tr>
    <tr>
      <th>1</th>
      <td>11</td>
      <td>10.147794</td>
      <td>95.884897</td>
      <td>1.000000</td>
      <td>0.586171</td>
    </tr>
    <tr>
      <th>2</th>
      <td>12</td>
      <td>13.850840</td>
      <td>170.770287</td>
      <td>0.913043</td>
      <td>0.351767</td>
    </tr>
    <tr>
      <th>3</th>
      <td>13</td>
      <td>5.661497</td>
      <td>33.993773</td>
      <td>0.960526</td>
      <td>0.406547</td>
    </tr>
    <tr>
      <th>4</th>
      <td>14</td>
      <td>2.815029</td>
      <td>13.459270</td>
      <td>1.000000</td>
      <td>0.461663</td>
    </tr>
    <tr>
      <th>5</th>
      <td>15</td>
      <td>4.812033</td>
      <td>31.726726</td>
      <td>1.000000</td>
      <td>0.508752</td>
    </tr>
  </tbody>
</table>
</div>


RMSE      6.912953\
Score    62.022741\
PICP      0.978928\
PINAW     0.470662

# DS03 (normal-filtered-degradation)


<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>unit</th>
      <th>RMSE</th>
      <th>Score</th>
      <th>PICP</th>
      <th>PINAW</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>10</td>
      <td>2.287758</td>
      <td>8.309579</td>
      <td>1.000000</td>
      <td>0.549712</td>
    </tr>
    <tr>
      <th>1</th>
      <td>11</td>
      <td>5.867247</td>
      <td>24.833282</td>
      <td>1.000000</td>
      <td>0.640541</td>
    </tr>
    <tr>
      <th>2</th>
      <td>12</td>
      <td>7.713505</td>
      <td>38.687416</td>
      <td>0.982143</td>
      <td>0.343562</td>
    </tr>
    <tr>
      <th>3</th>
      <td>13</td>
      <td>6.360628</td>
      <td>30.876775</td>
      <td>0.948276</td>
      <td>0.417969</td>
    </tr>
    <tr>
      <th>4</th>
      <td>14</td>
      <td>1.265477</td>
      <td>4.052291</td>
      <td>1.000000</td>
      <td>0.480910</td>
    </tr>
    <tr>
      <th>5</th>
      <td>15</td>
      <td>2.201658</td>
      <td>5.400241</td>
      <td>1.000000</td>
      <td>0.516345</td>
    </tr>
  </tbody>
</table>
</div>


RMSE      4.282712\
Score    18.693264\
PICP      0.988403\
PINAW     0.491506

## Old Results

## DS01

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>unit</th>
      <th>RMSE</th>
      <th>Score</th>
      <th>PICP</th>
      <th>PINAW</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>7</td>
      <td>5.775061</td>
      <td>55.267201</td>
      <td>0.999001</td>
      <td>0.309502</td>
    </tr>
    <tr>
      <th>1</th>
      <td>8</td>
      <td>3.785441</td>
      <td>57.762069</td>
      <td>1.000000</td>
      <td>0.295153</td>
    </tr>
    <tr>
      <th>2</th>
      <td>9</td>
      <td>6.018379</td>
      <td>99.245554</td>
      <td>1.000000</td>
      <td>0.428916</td>
    </tr>
    <tr>
      <th>3</th>
      <td>10</td>
      <td>5.708555</td>
      <td>92.050901</td>
      <td>0.998796</td>
      <td>0.415006</td>
    </tr>
  </tbody>
</table>
</div>

RMSE:	5.321859\
Score:	76.081431\
PICP:	0.999449\
PINAW:	0.362144

## DS03

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>unit</th>
      <th>RMSE</th>
      <th>Score</th>
      <th>PICP</th>
      <th>PINAW</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>10</td>
      <td>1.970763</td>
      <td>16.453508</td>
      <td>1.000000</td>
      <td>0.383870</td>
    </tr>
    <tr>
      <th>1</th>
      <td>11</td>
      <td>7.169923</td>
      <td>104.176352</td>
      <td>0.991818</td>
      <td>0.430604</td>
    </tr>
    <tr>
      <th>2</th>
      <td>12</td>
      <td>15.045545</td>
      <td>382.033190</td>
      <td>0.507948</td>
      <td>0.242492</td>
    </tr>
    <tr>
      <th>3</th>
      <td>13</td>
      <td>6.132815</td>
      <td>44.879145</td>
      <td>0.987355</td>
      <td>0.315902</td>
    </tr>
    <tr>
      <th>4</th>
      <td>14</td>
      <td>3.357752</td>
      <td>23.227942</td>
      <td>0.998947</td>
      <td>0.370075</td>
    </tr>
    <tr>
      <th>5</th>
      <td>15</td>
      <td>2.875275</td>
      <td>18.082909</td>
      <td>1.000000</td>
      <td>0.376594</td>
    </tr>
  </tbody>
</table>
</div>


RMSE:	6.092012\
Score:	98.142175\
PICP:	0.914345\
PINAW:	0.353256

## DS05 

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>unit</th>
      <th>RMSE</th>
      <th>Score</th>
      <th>PICP</th>
      <th>PINAW</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>7</td>
      <td>9.023385</td>
      <td>129.525538</td>
      <td>0.996078</td>
      <td>0.370195</td>
    </tr>
    <tr>
      <th>1</th>
      <td>8</td>
      <td>11.351902</td>
      <td>207.492181</td>
      <td>0.998291</td>
      <td>0.339061</td>
    </tr>
    <tr>
      <th>2</th>
      <td>9</td>
      <td>3.117663</td>
      <td>23.322696</td>
      <td>0.997016</td>
      <td>0.408783</td>
    </tr>
    <tr>
      <th>3</th>
      <td>10</td>
      <td>7.332727</td>
      <td>102.286471</td>
      <td>1.000000</td>
      <td>0.334772</td>
    </tr>
  </tbody>
</table>
</div>

RMSE:       7.706419\
Score:    115.656722\
PICP:       0.997846\
PINAW:      0.363203